naming/src/main/java/com/alibaba/nacos/naming/core/ServiceManager.java

```diff
@@ -637,14 +638,22 @@ private void putServiceAndInit(Service service) throws NacosException {
         Loggers.SRV_LOG.info("[NEW-SERVICE] {}", service.toJSON());
     }
 
-    public List<Service> searchServices(String namespaceId, String regex) {
+
+    public List<Service> searchServices(String namespaceId, String regex, boolean hasIpCount) {
+        if (StringUtils.isBlank(regex) && !hasIpCount) {
+            return new ArrayList<>(chooseServiceMap(namespaceId).values());
+        }
         List<Service> result = new ArrayList<>();
         for (Map.Entry<String, Service> entry : chooseServiceMap(namespaceId).entrySet()) {
             Service service = entry.getValue();
             String key = service.getName() + ":" + ArrayUtils.toString(service.getOwners());
-            if (key.matches(regex)) {
-                result.add(service);
+            if (StringUtils.isBlank(regex) || !key.matches(regex)) {
```

Comment:
Should be:
```java
            if (StringUtils.isNotBlank(regex) && !key.matches(regex)) {
                continue;
            }
```

Reply:
This case: when `Service Name is null` and `ipCount>0`,
I still take him to the outer layer, so the code looks very clear